src/dotnet/Notification.UI.Blazor/NotificationUI.cs

```diff
@@ -91,28 +94,25 @@ public Task RequestNotificationPermissions(CancellationToken cancellationToken)
         // Web browser notification permission requests are handled at notification-ui.ts
         => Task.CompletedTask;
 
-    [JSInvokable]
-    public Task HandleNotificationNavigation(string url)
-    {
+    public void DispatchNotificationNavigation(string url)
         // Called from MainActivity, i.e. unclear if it's running in Blazor Dispatcher
-        Dispatcher.InvokeAsync(() => {
-            var origin = UrlMapper.BaseUrl.TrimEnd('/');
-            if (url.IsNullOrEmpty() || !url.OrdinalStartsWith(origin))
-                return;
-
-            var chatPageRe = new Regex($"^{Regex.Escape(origin)}/chat/(?<chatid>[a-z0-9-]+)(?:#(?<entryid>)\\d+)?");
-            var match = chatPageRe.Match(url);
-            if (!match.Success)
-                return;
-
-            // Take relative URL to eliminate difference between web app and MAUI app
-            var relativeUrl = url[origin.Length..];
-
-            var chatIdGroup = match.Groups["chatid"];
-            if (chatIdGroup.Success)
-                History.NavigateTo(relativeUrl);
+        => _ = Dispatcher.InvokeAsync(async () => {
+            try {
+                await HandleNotificationNavigation(url);
+            }
+            catch (Exception e) {
+                Log.LogError(e, "Failed to dispatch notification navigation");
+            }
         });
-        return Task.CompletedTask;
+
+    [JSInvokable]
+    public async Task HandleNotificationNavigation(string url)
+    {
+        var relativeUrl = LocalUrl.FromAbsolute(url, UrlMapper);
+        if (relativeUrl?.IsChatId() == true) {
+            await History.NavigateTo(relativeUrl);
+            PanelsUI.Left.SetIsVisible(false);
```

Comment:
Should we set left panel invisible in wide mode?

Reply:
May be better make call like PanelsUI.Middle.EnsureVisible()

Reply:
> Should we set left panel invisible in wide mode?

it's handled in LeftPanel
https://github.com/Actual-Chat/actual-chat/blob/c6742a02be6b647254e2fe9d24662b4784d9f1a6/src/dotnet/UI.Blazor/Services/PanelsUI/LeftPanel.cs#L25